Update Asset Index to be BigInt

PJvGrol opened the pull request.

PJvGrol committed:
Update Asset Index to be BigInt

raymens committed:
Merge branch 'manual-improvements' into asset-index-bigint

raymens committed:
Update Algorand .NET SDK to 0.2.2.0-beta15

Updated the Algorand .NET SDK version from 0.2.2.0-beta14 to 0.2.2.0-beta15 to incorporate the latest features and bug fixes.

raymens committed:
Update Algorand .NET SDK to 0.2.2.0-beta16

Updated the Algorand .NET SDK version from 0.2.2.0-beta15 to 0.2.2.0-beta16 to incorporate the latest features and bug fixes.

raymens committed:
Update asset_id param type to ulong in SearchApi

Changed the parameter type for `asset_id` from `int?` to `ulong?` in the `AccountsAsync` and `TransactionsAsync` methods within the `SearchApi.cs` file, affecting both overloads of each method in the `Algorand.V2.Indexer` namespace.

raymens committed:
Update Algorand .NET SDK to 0.2.2.0-beta17

Updated the Algorand .NET SDK version from 0.2.2.0-beta16 to 0.2.2.0-beta17 to incorporate the latest features and bug fixes.

PJvGrol commented on dotnet-algorand-sdk/Transaction.cs:
```diff
@@ -314,7 +311,7 @@ public byte[] lease
         /// <param name="clawback">account which can issue clawbacks against holder accounts</param>
         private Transaction(Address sender, ulong? fee, ulong? firstValid, ulong? lastValid, byte[] note,
-                           string genesisID, Digest genesisHash, BigInteger? assetTotal, int assetDecimals, bool defaultFrozen,
+                           string genesisID, Digest genesisHash, ulong? assetTotal, int assetDecimals, bool defaultFrozen,
```
so it looks like it was ulong originally and we'd changed it to Bigint in an earlier stage. Why the switch back?

raymens replied:
the original (manual-improvements) implementation failed to create the transactions. It would try to write the BigIntiger as bytes and then failed to read it back. As there's no proper serializer defined for BigIntiger.

PJvGrol commented on dotnet-algorand-sdk/V2/Indexer/SearchApi.cs:
```diff
@@ -32,7 +32,7 @@ public partial interface ISearchApi
         /// <returns>(empty)</returns>
         /// <exception cref="ApiException">A server side error occurred.</exception>
-        System.Threading.Tasks.Task<Response> AccountsAsync(int? asset_id, int? limit, string next, ulong? currency_greater_than, bool? include_all, ulong? currency_less_than, string auth_addr, ulong? round, int? application_id);
+        System.Threading.Tasks.Task<Response> AccountsAsync(ulong? asset_id, int? limit, string next, ulong? currency_greater_than, bool? include_all, ulong? currency_less_than, string auth_addr, ulong? round, int? application_id);
```
if we're making assetid an ulong here, we should probably do the same on the actual asset model? (instead of bigint)

raymens replied:
we should =)

raymens committed:
Refine MetadataHash validation in Utils.cs

Removed the automatic generation of `asset.MetadataHash` in the Algorand namespace within Utils.cs. Now, the validation logic only allows a `MetadataHash` that is exactly 32 bytes long and throws an exception for any other length, ensuring that a valid hash must be explicitly provided.

raymens committed:
Update Algorand .NET SDK to 0.2.2.0-beta18

Updated the Algorand .NET SDK version from 0.2.2.0-beta17 to 0.2.2.0-beta18 to incorporate the latest features and bug fixes.

PJvGrol committed:
Update AssetTotal to be BigInteger

PJvGrol committed:
Update to beta 19

PJvGrol committed:
Change all assetIds to be ulong
